assets/src/adaptivity/rules-engine.ts

```diff
@@ -157,27 +157,27 @@ const processRules = (rules: JanusRuleProperties[], env: Environment) => {
         } else {
           actualValue = ogValue;
         }
-        const evaluatedValue = evaluateValueExpression(actualValue, env);
+        const evaluatedValue = evaluateValueExpression(actualValue, env, condition.id);
```

Comment:
Passing extra 'Key' parameter. They will be used in further improving the performance.